REPO-3902: Hotfix for RegistryCredentials & Ingress SG deletion functionality

amohammedalfresco opened the pull request:
- Refactored `helmHelperLambda` to make less api calls (especially for `ssm` to minimise `Resetting dropped connection: ssm.us-east-1.amazonaws.com` errors)
- Instead of downloading helper scripts separately, the whole directory containing scripts will be downloaded.
- Guard `getElb` script against non-empty values
- Implemented a logic to check for `RegistryCredentials` values.  If the parameter is set to `""` (empty value) then no `secret.yaml` will be created.  If set, the `base64` encoded value is tested first, if it is valid then will create the `secret.yaml` file, else fail the stack if the docker registry credentials value is not a valid `base64` encoded.
- Increased `helmHelperLambda` timeout to `180` to guard against Lambda getting timed out.
- `acs.yaml` cloudformation template reduced with helper scripts SSM Documents as it is not needed anymore.

gavincornwell commented on templates/acs-deployment-master.yaml:
```diff
@@ -371,9 +371,9 @@ Parameters:
+      Description: The private registry credentials base64.  Default value provided is an example in base64 encoded.
       NoEcho: True
-      Default: ""
+      Default: "ewoJImF1dGhzIjogewoJCSJodHRwczovL3ByaXZhdGVyZWdpc3RyeS5jb20iOiB7CgkJCSJhdXRoIjogImRHVnpkRUIwWlhOMExtTnZiVHB3WVhOemQyOXlaQW89IgoJCX0sCgl9LAoJIkh0dHBIZWFkZXJzIjogewoJCSJVc2VyLUFnZW50IjogIkRvY2tlci1DbGllbnQvMTguMDYuMS1jZSAoZGFyd2luKSIKCX0sCgkic3RhY2tPcmNoZXN0cmF0b3IiOiAia3ViZXJuZXRlcyIKfSUK"
```
This was set to an empty string to avoid committing something that looks like real credentials, see the comments on https://github.com/Alfresco/acs-deployment-aws/pull/11 for the discussion.

If this results in an error (it didn't in my testing) we should add Conditions to exclude the registry details when set to "".

amohammedalfresco replied:
I tried with literal string (ex: `false`) or leaving it as empty `""` but it throws below exception:
```
Error from server (BadRequest): error when creating "secret.yaml": Secret in version "v1" cannot be handled as a Secret: v1.Secret: ObjectMeta: v1.ObjectMeta: TypeMeta: Kind: Data: decode base64: illegal base64 data at input byte 0, error found in #10 byte of ...|--install"},"kind":"|..., bigger context ...|sion":"v1","data":{".dockerconfigjson":"--install"},"kind":"Secret","metadata":{"name":"quay-registr|...
```

amohammedalfresco replied:
I will try to condition the `RegistryCredential` so if it is set to empty value `""`, don't create the secret.yaml, else create the secret.

gavincornwell replied:
Yes, that would be ideal!

gavincornwell commented on templates/acs.yaml:
```diff
@@ -138,9 +138,9 @@ Parameters:
+      Description: The private registry credentials base64.  Default value provided is an example in base64 encoded.
       NoEcho: True
-      Default: false
+      Default: "ewoJImF1dGhzIjogewoJCSJodHRwczovL3ByaXZhdGVyZWdpc3RyeS5jb20iOiB7CgkJCSJhdXRoIjogImRHVnpkRUIwWlhOMExtTnZiVHB3WVhOemQyOXlaQW89IgoJCX0sCgl9LAoJIkh0dHBIZWFkZXJzIjogewoJCSJVc2VyLUFnZW50IjogIkRvY2tlci1DbGllbnQvMTguMDYuMS1jZSAoZGFyd2luKSIKCX0sCgkic3RhY2tPcmNoZXN0cmF0b3IiOiAia3ViZXJuZXRlcyIKfSUK"
```
Agree this should be consistent but should be an empty string, not bogus credentials.

amohammedalfresco commented:
- Refactored `helmHelperLambda` to make less api calls (especially for `ssm` to minimise `Resetting dropped connection: ssm.us-east-1.amazonaws.com` errors)
- Instead of downloading helper scripts separately, the whole directory containing scripts will be downloaded.
- Guard `getElb` script against non-empty values
- Implemented a logic to check for `RegistryCredentials` values.  If the parameter is set fo `""` then no `secret.yaml` will be created.  If set, the `base64` encoded value is tested first, if it is valid then will create the `secret.yaml` file, else (either we can raise a warning or if decided can fail the stack [TBD])
- Increased `helmHelperLambda` timeout to `180` to guard against Lambda getting timed out.
- `acs.yaml` cloudformation template reduced with helper scripts SSM Documents as it is not needed anymore.

gavincornwell commented on templates/acs-deployment-master.yaml:
```diff
@@ -371,7 +371,7 @@ Parameters:
     RegistryCredentials:
       Type: String
-      Description: The private registry credentials base64
+      Description: The private registry credentials with base64 encoded.
```
I would say: "The private registry credentials in base64 format."

gavincornwell commented on templates/acs.yaml:
```diff
@@ -138,9 +138,9 @@ Parameters:
     RegistryCredentials:
       Type: String
-      Description: The private registry credentials base64
+      Description: The private registry credentials with base64 encoded.
```
I would say: "The private registry credentials in base64 format."
